### ISSUE TITLE
Superclass Constructor Mocking not working in 5.14.2 

### ISSUE THREAD

**Issue by ThomasH-75** [mockito/mockito#3521]
Hi,

Mocking the construction of a superclass, and having the mock kick-in when instantiating the child-class worked in **mockito-inline v4.6.1**, but not with: **mockito-core v5.14.2**

The following TestNG unit test passes with **4.6.1** and fails with **5.14.2**.
The failure is simply that getInt() returns 0, so the assert fails.

    import org.mockito.MockedConstruction;
    import org.mockito.Mockito;
    import org.testng.Assert;
    import org.testng.annotations.Test;

    public class NewTest {
       @Test
       public void test() {
           try (MockedConstruction<Super> superConstructorMock = Mockito.mockConstruction(Super.class,
                   (mock, context) -> {
                       Mockito.when(mock.getInt()).thenReturn(42);
                   })) {

               Child c = new Child();
               Assert.assertEquals(c.getInt(), 42);
           }
       }

       class Super {
           int getInt() {
               return 0;
           }
       }

       class Child extends Super {}
    }

I have tried both Java 17 and 21, and get the same behavior.
I have run it with Maven and in Eclipse, on Linux.

/Thomas

**Comment by ccarbonellbh** [comment]
I'm running into the same problem when trying to update to 5.14.2. Having trouble finding any details about what exactly the cause might be.

**Comment by nikialeksey** [comment]
Hi!

I created a similar ticket here #3525 , and I found that the problem was introduced here: https://github.com/mockito/mockito/commit/02615ffb8e1977296ef66e445a09d7e6611898ea by @raphw .

**Comment by raphw** [comment]
We could possibly offer a convenience:

    public static <T> MockedConstruction<T> mockConstruction(Class<T> classToMock, Class<? extends T>... subclasses);

where one can also specify the mocking of subclasses in one line. It's hard to discover all possible subclasses, though, as they might be loaded dynamically, so I think some explicit API would be mandated.

**Comment by nikialeksey** [comment]
I disagree that this is not a bug. My[not a mockito developer, but mockito client] understanding is next. 

Let's say we have a `class Child extend Super`. When we instantiate `Child` class bytecode of his constructor will actually execute `Super`'s constructor, right? Then I create a `mockConstruction` for `Super`, and I expect that Super constructor will not be called (because I made a mockConstruction for it), but it is actually called (as I described in #3525).

I actually okay with idea to have 

```java
public static <T> MockedConstruction<T> mockConstruction(Class<T> classToMock, Class<? extends T>... subclasses);
```

It will allow me to update mockito version in my project (because I have a lot of tests which depends on this logic).
